include/seqan3/range/concept.hpp

```diff
@@ -132,6 +132,33 @@ template <typename type>
 concept bool random_access_range_concept = bidirectional_range_concept<type> && (bool)ranges::RandomAccessRange<type>();
 //!\endcond
 
+/*!\interface seqan3::const_iterable_concept <>
+ * \extends seqan3::input_range_concept
+ * \brief Specifies requirements of an input range type for which the `const` version of that type satisfies the
+ * same strength input range concept as the non-const version.
+ *
+ * \details
+ *
+ * For a type `t` it usually holds that if `t` is a range, `t const` is also a range with similar properties, but
+ * there are cases where this does not hold:
+ *
+ *   * a `const` range is usually not writable so seqan3::output_range_concept is lost; pure output ranges
+ * (those that are not also input ranges) are therefore not `const`-iterable;
+ *   * single-pass input ranges, like SeqAn files, are not `const`-iterable, because "single-pass-ness" implies that
+ * there is something in the range that changes on every iterator increment (and `const` ranges can't change);
+ *   * certain views store a state with their algorithm that also changes when `begin()` is called or an
+ * iterator is incremented; these may be not be `const`-iterable, because the standard library
+ * (and also SeqAn3) guarantees that it is safe to call `const`-qualified functions concurrently.
+ */
+//!\cond
+template <typename type>
+concept bool const_iterable_concept =
+    input_range_concept<std::remove_const_t<type>>          && input_range_concept<type const> &&
+    (forward_range_concept<std::remove_const_t<type>>       == forward_range_concept<type const>) &&
+    (bidirectional_range_concept<std::remove_const_t<type>> == bidirectional_range_concept<type const>) &&
+    (random_access_range_concept<std::remove_const_t<type>> == random_access_range_concept<type const>);
```

Comment:
In this version, only random_access_ranges would be const_iterable. You probably want to check that any of the expressions is true not all?

Reply:
I don't understand what you mean, but maybe you also didn't understand the code? Observe the parentheses :wink: